lib/api/api.js

```diff
@@ -100,34 +101,35 @@ const api = {
                 log.trace('authentication error', { error: err });
                 return callback(err);
             }
-            if (authorizationResults) {
+            function ifAuthResults(authResults) {
```

Comment:
Just curious, what is this way of declaring a function?
Like are we declaring it inside the variable api?
Never seen this trick. How does this work?